modules/Core/pages/forgot_password.php

```diff
@@ -23,9 +23,20 @@
     // Enter email address form
     if (Input::exists()) {
         if (Token::check()) {
-            if (!isset($_POST['email']) || empty($_POST['email'])) {
-                $error = $language->get('user', 'email_required');
-            } else {
+
+            $validation = Validate::check([
+                'email' => [
+                    Validate::REQUIRED => true,
+                    Validate::RATE_LIMIT => [1, 60] // 1 attempt every 60 seconds
+                ]
+            ])->messages([
+                'email' => [
+                    Validate::REQUIRED => $language->get('user', 'email_required'),
+                    Validate::RATE_LIMIT => fn($meta) => $language->get('general', 'rate_limit', $meta)
```

Comment:
This could be a static function (see aber's changes to login.php in his PR: https://github.com/NamelessMC/Nameless/pull/3128/files#diff-4be83c7d87065b9b6e0404325a3c8d6a33f61b96904d4686c0280bdeaa61c5fa)